pkg/util/resizefs/resizefs_linux.go

```diff
@@ -84,3 +100,20 @@ func (resizefs *ResizeFs) xfsResize(deviceMountPath string) (bool, error) {
 	resizeError := fmt.Errorf("resize of device %s failed: %v. xfs_growfs output: %s", deviceMountPath, err, string(output))
 	return false, resizeError
 }
+
+// findBlockDeviceRescanPath Find the underlaying disk for a linked path such as /dev/disk/by-path/XXXX or /dev/mapper/XXXX
+// will return /sys/devices/pci0000:00/0000:00:15.0/0000:03:00.0/host0/target0:0:1/0:0:1:0/rescan
+func findBlockDeviceRescanPath(path string) (string, error) {
+	devicePath, err := filepath.EvalSymlinks(path)
+	if err != nil {
+		return "", err
+	}
+	// if path /dev/hdX split into "", "dev", "hdX" then we will
+	// return just the last part
+	parts := strings.Split(devicePath, "/")
+	if len(parts) == 3 && strings.HasPrefix(parts[1], "dev") {
+		blockPath := "/sys/block" + "/" + parts[2] + "/device/rescan"
```

Comment:
IMO, we define some consts to replace “xx”，moreover，filepath.join may be better for building the path.

Reply:
done